third_party/blink/renderer/core/farbling/brave_session_cache.cc

```diff
@@ -263,8 +245,23 @@ void BraveSessionCache::Init() {
 }
 
 void BraveSessionCache::FarbleAudioChannel(float* dst, size_t count) {
-  if (audio_farbling_helper_)
-    audio_farbling_helper_->FarbleAudioChannel(dst, count);
+  if (settings_client_ == nullptr) {
+    return;
+  }
+  const auto audio_farbling_level = settings_client_->GetBraveFarblingLevel(
```

Comment:
you're going to introduce a huge performance hit with this change. You should not re-check the farbling level when `audio_farbling_helper_` is already created.

pls be careful to not slowdown other things in a similar way.

Reply:
fixed